### PR TITLE
statistics: enables global-level stats to be generated in fast analyze mode

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

sub PR for [#22554](https://github.com/pingcap/tidb/pull/22554)

Problem Summary:
1. Generate the global-level stats when we analyze the partition table in `fast analyze mode`.
2. Fix an issue for `globalStats.Count`. 

### What is changed and how it works?
Same as described above.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -171,6 +171,12 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			sc := e.ctx.GetSessionVars().StmtCtx
 			globalStats, err := statsHandle.MergePartitionStats2GlobalStats(sc, infoschema.GetInfoSchema(e.ctx), globalStatsID.tableID, info.isIndex, info.idxID)
 			if err != nil {
+				errMessage := err.Error()
+				if errMessage == "[stats] build global-level stats failed due to missing partition-level stats" {
```
Comment (qw4990) [inline executor/analyze.go:175]: Judging an error by comparing its error message is dangerous.
You'd better create a new kind of error and use `errors.Cause(err)`.
Here is an example:
```
			if _, ok := errors.Cause(err).(*tikv.PDError); !ok {
				break
			}
```

File: executor/analyze.go [hunk 2]
```diff
@@ -809,8 +815,7 @@ func (e *AnalyzeFastExec) calculateEstimateSampleStep() (err error) {
 		sql := new(strings.Builder)
 		sqlexec.MustFormatSQL(sql, "select count(*) from %n.%n", dbInfo.Name.L, e.tblInfo.Name.L)
 
-		pruneMode := variable.PartitionPruneMode(e.ctx.GetSessionVars().PartitionPruneMode.Load())
-		if pruneMode != variable.DynamicOnly && e.tblInfo.ID != e.tableID.GetStatisticsID() {
+		if e.tblInfo.ID != e.tableID.GetStatisticsID() {
```
Comment (rebelice) [inline executor/analyze.go:818]: Why don't we check `PartitionPruneMode`

**Review by qw4990 — COMMENTED** [review]
LGTM

**Review by qw4990 — COMMENTED** [review]
/LGTM

File: executor/analyze.go (hunk 2 above)
Comment (Reminiscent) [reply executor/analyze.go:817]: The `AnalyzeFastExec` structure is based on partition. The content recorded in it corresponds to the content of the partition, not the whole table.

**Review by rebelice — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]
/LGTM

**Pull request merged**